private/bufpkg/bufconnect/interceptors.go

```diff
@@ -112,16 +112,27 @@ func NewAuthorizationInterceptorProvider(tokenProviders ...TokenProvider) func(s
 		interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
 			return connect.UnaryFunc(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
 				usingTokenEnvKey := false
+				hasToken := false
 				for _, tf := range tokenProviders {
 					if token := tf.RemoteToken(address); token != "" {
 						req.Header().Set(AuthenticationHeader, AuthenticationTokenPrefix+token)
 						usingTokenEnvKey = tf.IsFromEnvVar()
+						hasToken = true
 						break
 					}
 				}
 				response, err := next(ctx, req)
-				if err != nil && usingTokenEnvKey {
-					err = &AuthError{cause: err, tokenEnvKey: tokenEnvKey}
+				if err != nil {
```

Comment:
Perhaps it should only wrap it in an `AuthError` when the return code is `CodeUnauthenticated` or `isEmptyUnknownError(err)`?

Reply:
This expects the caller to handle the error code check -- this pattern seems reasonable to me.

Reply:
In this case, since this is the auth interceptor, this behaviour makes sense because from this interceptor's perspective, it is always returning an `AuthError`.